sklearnex/basic_statistics/basic_statistics.py

```diff
@@ -14,4 +14,130 @@
+
+from .._device_offload import dispatch
+from .._utils import PatchingConditionsChain
+
+
+@control_n_jobs(decorated_methods=["fit"])
+class BasicStatistics(BaseEstimator):
+    """
+    Estimator for basic statistics.
+    Allows to compute basic statistics for provided data.
+    Parameters
+    ----------
+    result_options: string or list, default='all'
+        List of statistics to compute
+
+    Attributes (are existing only if corresponding result option exists)
+    ----------
+        min : ndarray of shape (n_features,)
+            Minimum of each feature over all samples.
+        max : ndarray of shape (n_features,)
+            Maximum of each feature over all samples.
+        sum : ndarray of shape (n_features,)
+            Sum of each feature over all samples.
+        mean : ndarray of shape (n_features,)
+            Mean of each feature over all samples.
+        variance : ndarray of shape (n_features,)
+            Variance of each feature over all samples.
+        variation : ndarray of shape (n_features,)
+            Variation of each feature over all samples.
+        sum_squares : ndarray of shape (n_features,)
+            Sum of squares for each feature over all samples.
+        standard_deviation : ndarray of shape (n_features,)
+            Standard deviation of each feature over all samples.
+        sum_squares_centered : ndarray of shape (n_features,)
+            Centered sum of squares for each feature over all samples.
+        second_order_raw_moment : ndarray of shape (n_features,)
+            Second order moment of each feature over all samples.
+    """
+
+    def __init__(self, result_options="all"):
+        self.options = result_options
+
+    _onedal_basic_statistics = staticmethod(onedal_BasicStatistics)
+
+    def _save_attributes(self):
+        assert hasattr(self, "_onedal_estimator")
+
+        if self.options == "all":
+            result_options = onedal_BasicStatistics.get_all_result_options()
+        else:
+            result_options = self.options
+
+        if isinstance(result_options, str):
+            setattr(self, result_options, getattr(self._onedal_estimator, result_options))
+        elif isinstance(result_options, list):
+            for option in result_options:
+                setattr(self, option, getattr(self._onedal_estimator, option))
+
+    def _onedal_supported(self, method_name, *data):
+        patching_status = PatchingConditionsChain(
+            f"sklearnex.basic_statistics.{self.__class__.__name__}.{method_name}"
+        )
+        return patching_status
+
+    _onedal_cpu_supported = _onedal_supported
+    _onedal_gpu_supported = _onedal_supported
+
+    def _onedal_fit(self, X, sample_weight=None, queue=None):
+        if sklearn_check_version("1.0"):
+            X = self._validate_data(X, dtype=[np.float64, np.float32], ensure_2d=False)
+        else:
+            X = check_array(X, dtype=[np.float64, np.float32], ensure_2d=False)
```

Comment:
```suggestion
        X = self._validate_data(X, dtype=[np.float64, np.float32], ensure_2d=False)
```
We are not targeted to support sklearn < 1.0

Reply:
then I guess we need to remove it across the whole repo. I'll create the ticket.

Reply:
Thank you! Addressed https://github.com/intel/scikit-learn-intelex/pull/1906